### PR TITLE
Feature/ak-33: Added AWS Unused ENI plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/ec2/unusedEni.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Unused Elastic Network Interfaces',
+    category: 'EC2',
+    description: 'Ensures that unused AWS Elastic Network Interfaces (ENIs) are removed.',
+    more_info: 'Unused AWS ENIs should be removed to follow best practices and to avoid reaching the service limit.',
+    recommended_action: 'Delete the unused AWS Elastic Network Interfaces',
+    link: 'https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html',
+    apis: ['EC2:describeNetworkInterfaces', 'STS:getCallerIdentity'],
```
Comment (dipsubha06) [inline plugins/aws/ec2/unusedEni.js:11]: STS:getCallerIdentity is not being used. It can be removed then.

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

File: plugins/aws/ec2/unusedEni.js (hunk 1 above)
Comment (AkhtarAmir) [reply plugins/aws/ec2/unusedEni.js:11]: It is being used at line 20

File: plugins/aws/ec2/unusedEni.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+            if(!describeNetworkInterfaces.data.length) {
+                helpers.addResult(results, 0, 'No AWS ENIs found', region);
+                return rcb();
+            }
+
+            describeNetworkInterfaces.data.forEach(function(eni){
+                var resource = `arn:${awsOrGov}:ec2:${region}:${accountId}:network-interface/${eni.NetworkInterfaceId}`;
+
+                if (eni.Status === 'in-use') {
```
Comment (giorod3) [inline plugins/aws/ec2/unusedEni.js:42]:
```suggestion
                if (eni.Status && eni.Status === 'in-use') {
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
just one comment here

File: plugins/aws/ec2/unusedEni.js (hunk 2 above)
Comment (AkhtarAmir) [reply plugins/aws/ec2/unusedEni.js]: Updated.

**Review by dipsubha06 — APPROVED** [review]

File: plugins/aws/ec2/unusedEni.js (hunk 2 above)
Comment (giorod3) [reply plugins/aws/ec2/unusedEni.js]: @AkhtarAmir I don't see this updated in the PR

**Review by giorod3 — APPROVED** [review]

**Pull request merged**